scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/SortModifiers.scala

```diff
@@ -0,0 +1,92 @@
+package org.scalafmt.rewrite
+
+import org.scalafmt.config.SortSettings._
+
+import scala.meta.Tree
+import scala.meta._
+
+object SortModifiers extends Rewrite {
+
+  override def rewrite(code: Tree, ctx: RewriteCtx): Seq[Patch] = {
+    val order = ctx.style.rewrite.sortModifiers.order
+
+    val sortMods: Seq[Mod] => Seq[Mod] = { mods =>
+      mods.sortWith(orderModsBy(order))
+    }
+
+    /*
+     * in the case of Class, Object, and of class constructor parameters
+     * some Mods are immovable, e.g. "case class X".
+     *
+     * The case of parameters is a bit more curious, because there the
+     * "val" or "var" in, say:
+     * {{{
+     *   class Test(private final val x: Int)
+     * }}}
+     * are considered Mods, instead of being similar to `Defn.Val`, or `Defn.Var`.
+     */
+    val patchesOfPatches = code.collect {
+      case d: Defn.Def => patchMods(sortMods, d.mods)
+      case v: Defn.Val => patchMods(sortMods, v.mods)
+      case v: Defn.Var => patchMods(sortMods, v.mods)
+      case t: Defn.Type => patchMods(sortMods, t.mods)
+      case c: Defn.Class => patchMods(sortMods, c.mods.filterNot(isCase))
+      case o: Defn.Object => patchMods(sortMods, o.mods.filterNot(isCase))
+      case t: Defn.Trait => patchMods(sortMods, t.mods)
+      case p: Term.Param => patchMods(sortMods, p.mods.filterNot(isValOrVar))
+    }
+    patchesOfPatches.flatten
+  }
+
+  private val isValOrVar: Mod => Boolean = m =>
+    m.is[Mod.ValParam] || m.is[Mod.VarParam]
+
+  private val isCase: Mod => Boolean = m => m.is[Mod.Case]
+
+  private def patchMods(
+      sortMods: Seq[Mod] => Seq[Mod],
+      oldMods: Seq[Mod]): Seq[Patch] = {
+    if (oldMods.isEmpty) Nil
+    else {
+      val sortedMods: Seq[Mod] = sortMods(oldMods)
+      sortedMods.zip(oldMods).flatMap {
+        case (next, old) =>
+          if (old.tokens.isEmpty) {
+            //Not sure why this happens, how can you have a mod with no tokens?
```

Comment:
Yes, there is a bug for example for `def foo(implicit x: Int)`, the `implicit` has no tokens. The user can't do anything about it so it's fine to ignore it.

Reply:
Please remove the comment, this guard is totally fine.